### PR TITLE
Feature 728 new TCdata

### LINKED ISSUE #728: Apply TCPairs, TCStat, CyclonePlotter, GridStat to EMC 2020 Data
*Replace italics below with details for this issue.*

## Describe the Task ##
*Learn how to use TCPairs, TCStat, CyclonePlotter, and GridStat in the context of cyclones, using existing sample data from GitHub*

### Time Estimate ###
*Estimate the amount of work required here.*
*3 calendar days of work spread out over December.*
*Future work will either extend this task or create another one*

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
* John O will provide more data probably in January.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [x] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [ ] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by fentoad72**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
- [ ] End to end wrapper with a python script that processes storm data into METplus_output/decks/ADECK & BDECK files, runs TC_Pairs_extra_tropical.conf on the ADECK & BDECK creating METplus_output/tc_pairs.tcst file, then runs CyclonePlotter to generate a map of storm tracks; all of this is centered on a single YYYYMMDDHH specified in the Cyclone_fcstGFS_obsGFS_OPC.conf wrapper.  Tests have been run for 2020120700 and 2020100700 <br>
- [ ] Other data exists on kiowa in /d2/projects/extra-tc_verif

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
- [ ] do end to end testing using 2020100700 and maybe one or a few other dates
- [ ] if applicable try using other data not included in METplus_Data (in /d2/projects/extra-tc_verif)

- [x] Do these changes include sufficient documentation and testing updates? **[Yes or No]**
New use case (CyclonePlotter_fcstGFS_obsGFS_OPC.conf) with its own use case page in model_applications/tc_and_extra_tc
Testing is included in the internal tests suite

- [x] Will this PR result in changes to the test suite? **[Yes or No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>
Yes.  Output includes ADECK & BDECK files, tc_pairs.tcst, and YYYYMMDD.png and txt

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, and **Milestone**
- [ ] After submitting the PR, select **Linked Issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: metplus/wrappers/cyclone_plotter_wrapper.py
Comment (jprestop) [inline metplus/wrappers/cyclone_plotter_wrapper.py]: Do we need to keep these commented out lines here or should we remove them?

**Review by jprestop — COMMENTED** [review]
Other .py files in similar subdirectory structures under parm/use_cases/, use only one "#" symbol for comments.  Please change the comments in parm/use_cases/model_applications/tc_and_extra_tc/CyclonePlotter_fcstGFS_obsGFS_OPC/extract_opc_decks.py to be consistent with the style in other similar files.  For example, please change
`### column names/dictionary keys for the trak.data file`
to  
`# column names/dictionary keys for the trak.data file`
etc.

**Review by jprestop — COMMENTED** [review]
Also, I see:
"Some checks were not successful
34 successful, 1 failing, and 1 skipped checks"

Please check on this and reiterate until all checks are successful.

File: metplus/wrappers/cyclone_plotter_wrapper.py
Comment (fentoad72) [reply metplus/wrappers/cyclone_plotter_wrapper.py]: I removed the legend because it was too large and obscured the plot.  I just tested with these lines uncommented and the legend is no longer covering the plot area, so I'll uncomment them (they are normally uncommented)

Comment (fentoad72) [reply metplus/wrappers/cyclone_plotter_wrapper.py]: These lines were commented out by me because the legend became so large it obscured the figure for feature_728_newTCdata.  Added them back in and tested; the legend no longer obscures the figure.

**Review by jprestop — COMMENTED** [review]
On kiowa, in /d1/projects/METplus/METplus_pull_requests/METplus-4.0.0-beta4/feature_728/METplus I ran:

> bash
> conda deactivate
> conda activate /home/met_test/.conda/envs/
> run_metplus.py -c parm/use_cases/model_applications/tc_and_extra_tc/CyclonePlotter_fcstGFS_obsGFS_OPC.conf -c /d1/projects/METplus/METplus_pull_requests/METplus-4.0.0-beta4/feature_728/jpresto.kiowa.conf

which produced some information about deprecations:
> /home/met_test/.conda/envs/metplus_dev/lib/python3.6/site-packages/cartopy/mpl/gridliner.py:307: UserWarning: The .xlabels_top attribute is deprecated. Please use .top_labels to toggle visibility instead.
  warnings.warn('The .xlabels_top attribute is deprecated. Please '
/home/met_test/.conda/envs/metplus_dev/lib/python3.6/site-packages/cartopy/mpl/gridliner.py:331: UserWarning: The .ylabels_left attribute is deprecated. Please use .left_labels to toggle visibility instead.

and an error:

> 02/25 11:29:35.123 metplus.CyclonePlotter (met_util.py:214) ERROR: UserScript had 1 error.
02/25 11:29:35Z run-METplus-metplus.CyclonePlotter: ERROR:  UserScript had 1 error.

Looking in /d1/projects/METplus/METplus_pull_requests/METplus-4.0.0-beta4/feature_728/out/logs/metplus.log.20210225112931, I see:

> FileNotFoundError: [Errno 2] File /d1/projects/METplus/METplus_Data/model_applications/tc_and_extra_tc/CyclonePlotter_fcstGFS_obsGFS_OPC/trak.gfso.atcf_gen.glbl.2020 does not exist: '/d1/projects/METplus/METplus_Data/model_applications/tc_and_extra_tc/CyclonePlotter_fcstGFS_obsGFS_OPC/trak.gfso.atcf_gen.glbl.2020'

Should the data be in /d1/projects/METplus/METplus_Data/ also?  My apologies.  I am not very familiar with what is supposed to be where.

When I rerun using the INPUT_BASE as /d2/projects/extra-tc_verif/, I see in /d1/projects/METplus/METplus_pull_requests/METplus-4.0.0-beta4/feature_728/out/logs/metplus.log.20210225113937, a similar error:

> FileNotFoundError: [Errno 2] File /d2/projects/extra-tc_verif/model_applications/tc_and_extra_tc/CyclonePlotter_fcstGFS_obsGFS_OPC/trak.gfso.atcf_gen.glbl.2020 does not e
xist: '/d2/projects/extra-tc_verif/model_applications/tc_and_extra_tc/CyclonePlotter_fcstGFS_obsGFS_OPC/trak.gfso.atcf_gen.glbl.2020'

I see the data here:
> /d2/projects/extra-tc_verif/gpfs/dell1/nco/ops/com/gentracks/prod/gentracks/2020/trak.gfso.atcf_gen.glbl.2020

where can I test this so that the data is relative to {INPUT_BASE} and the directory structure of the use case?

Also, regarding:
> do end to end testing using 2020100700 and maybe one or a few other dates

I wouldn't know what other dates to use.  If what is in the conf file is not what I should test, please provide an additional conf file for me to test with or indicate what I should change in the current conf file.

>  if applicable try using other data not included in METplus_Data (in /d2/projects/extra-tc_verif)

I'm not sure what is meant by "if applicable".  If other testing is needed, please let me know more specifically what should be tested.

**Comment by georgemccabe** [comment]
Sorry, @jprestop , I forgot to put the data in the METplus_Data development directory. I just moved it here:

/d1/projects/METplus/METplus_Data/development/feature_728_newTCdata

So you can set your INPUT_BASE to that directory and it should run.

**Comment by fentoad72** [comment]
On Kiowa, the EMC data is all in /d2/projects/extra-tc_verif/gpfs/dell1/nco/ops/com/gentracks/prod/gentracks/2020; the file used is trak.gfso.atcf_gen.glbl.2020  @georgemccabe uploaded that file to Mohawk under met_test; looks like he commented above on how to find it.  You can disregard what I said about testing other dates or files; I wasn't sure to what extent testing was done.

**Comment by jprestop** [comment]
Thanks @georgemccabe for putting the data in the METplus_Data area.  Thanks @fentoad72 for the update on testing.

Setting INPUT_BASE to /d1/projects/METplus/METplus_Data/development/feature_728_newTCdata produced a successful run.  I checked only to make sure there were output files and noticed that one of the output file names contains two extensions ".tcst.tcst":
`(metplus_dev) jpresto@kiowa:/d1/projects/METplus/METplus_pull_requests/METplus-4.0.0-beta4/feature_728/METplus$ ls ../out/tc_pairs/
tc_pairs.2020100700.tcst.tcst
`
Is that what we want or should there only be one .tcst?

Also, does anything need to be changed due to the deprecation warnings?

> /home/met_test/.conda/envs/metplus_dev/lib/python3.6/site-packages/cartopy/mpl/gridliner.py:307: UserWarning: The .xlabels_top attribute is deprecated. Please use .top_labels to toggle visibility instead.
warnings.warn('The .xlabels_top attribute is deprecated. Please '
/home/met_test/.conda/envs/metplus_dev/lib/python3.6/site-packages/cartopy/mpl/gridliner.py:331: UserWarning: The .ylabels_left attribute is deprecated. Please use .left_labels to toggle visibility instead.

**Comment by fentoad72** [comment]
The deprecation warnings come from running CyclonePlotter.conf using cartopy.  CyclonePlotter was meant to be a quicklook at tc_pairs output resembling the NCEP figures of storm data (https://www.emc.ncep.noaa.gov/mmb/gplou/emchurr/glblgen/index.html) and I'm not sure what the plans are with it.  Minna would know more (@bikegeek?)

**Comment by fentoad72** [comment]
I've got green checkmarks for all 36 checks and there is no conflict wit the base branch.  There are 71 formatting errors in the documentation (having to do with files I didn't modify).  Is this ready to be merged or is there additional work I need to do?

**Comment by georgemccabe** [comment]
If Julie approves then I think it is ready to merge

**Review by jprestop — APPROVED** [review]
I was waiting for Minna's reply @bikegeek . I do approve this request though.  **Please take a look at this following webpage and follow the instructions from "8.6.2 Merge the pull request and ensure that all tests pass" and below.**  8.6.2 currently falls under "8.6 Pull Request Reviewer Instructions", but I will be moving them to a "8.7 After the Pull Request is Approved" section since it is now the requestor and no longer the reviewer that merges.  
http://icculus.rap.ucar.edu/metplus-contrib/Contributors_Guide/add_use_case.html

**Comment by fentoad72** [comment]
Merged 728 into develop and deleted feature_728_newTCdata branch

**Pull request merged**
